### ISSUE TITLE
The Tag component remove button needs to be keyboard accessible

### ISSUE THREAD

**Issue by nicklemmon** [SparkPost/matchbox#197]
The `<Tag/>` component remove button is not currently keyboard accessible.

## Requirements

* The remove button should be focusable by keyboard users
* The remove button should be interactive with the enter key and spacebar

## Potential Fixes

* Update the HTML to use the `<button>` element instead of the `<a>`
* __or__ use the existing `<a>` element and add `role="button"` alongside a relevant value for `href` - `href="javascript:void(0):"` should do the trick
